cmd/destroy/runners.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destroy
+
+import (
+	"context"
+
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/cmd/pipeline"
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// runnerI defines the different functions to run okteto inside an okteto deploy
+// or an okteto destroy directly
+type runnerI interface {
+	translateConfigMapAndDeploy(context.Context, *pipeline.CfgData, kubernetes.Interface) (*apiv1.ConfigMap, error)
+	destroyConfigMap(context.Context, *apiv1.ConfigMap, string, kubernetes.Interface) error
+	setErrorStatus(context.Context, *apiv1.ConfigMap, *pipeline.CfgData, error, kubernetes.Interface) error
+}
+
+// oktetoInsideOktetoRunner is the runner used when the okteto is executed
+// inside an okteto deploy command
+type oktetoInsideOktetoRunner struct{}
```

Comment:
Same about `okteto` prefix. Also, I don't think `insideOkteto` is a good name or it can be misleading. What does inside Okteto means? Maybe we can name them `standalone` and `subcommand`.

Other option could be to name them according to what they really do (modify or not the config map) instead of setting the names based on the condition to use one or the other, but nothing comes to my mind